src/coreComponents/codingUtilities/TableData.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableData.hpp
+ */
+
+#ifndef GEOS_COMMON_TableData_HPP
+#define GEOS_COMMON_TableData_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/Format.hpp"
+
+namespace geos
+{
+
+// Class for managing table data
+class TableData
+{
+public:
+
+  /**
+   * @brief Add a row to the table.
+   * @param Args The values passed to addRow (can be any type).
+   * @param args Cell values to be added to the row.
+   */
+  template< typename ... Args >
+  void addRow( Args const & ... args );
+
+  /**
+   * @brief Add a row to the table
+   * @param row A vector of string who contains cell Values
+   */
+  void addRow( std::vector< string > const & row );
+
+  /**
+   * @brief Reset data in the table
+   */
+  void clear();
+
+  /**
+   * @return The rows of the table
+   */
+  std::vector< std::vector< string > > const & getTableDataRows() const;
+
+private:
+
+  std::vector< std::vector< string > > m_rows;
+
+};
+
+// Class for managing 2D table m_data
+class TableData2D
+{
+public:
+
+  /**
+   * @brief Add a cell to the table.
+   * Construct a map of pair<x,y> and cell value
+   * @param T The value passed to addCell (can be any type).
+   * @param value Cell value to be added.
+   */
+  template< typename T >
+  void addCell( real64 rowValue, real64 columnValue, T const & value );
+
+  /**
+   * @brief Construct a TableData from a Table2D
+   * @return A TableData
+   */
+  TableData buildTableData() const;
+
+  /**
+   * @return return all columns values for 2D table
+   */
+  std::set< real64 > const & getColumns() const;
+
+  /**
+   * @return return all rows values for 2D table
+   */
+  std::set< real64 > const & getRows() const;
+
+private:
+  std::map< std::pair< real64, real64 >, string > m_data;
+  std::set< real64 > m_columns;
+  std::set< real64 > m_rows;
```

Comment:
Doing so, aren't you losing the ordering of your insertions? Is this intended?

Reply:
yes, it's intended as a 2D Layout suggests increasing columns / rows.
I'm adding documentation to indicate this behavior

Reply:
> 2D Layout suggests increasing columns / rows

Probably you know better than  me. Be sure to document it then.